src/main/java/de/teamlapen/vampirism/entity/EntityVampirism.java

```diff
@@ -348,11 +353,57 @@ protected void updateAITasks() {
      * @param actionstmp
      *            list of action sets
      */
-    protected List<IEntityAction> getAvailableActions(List<IEntityAction[]> actionstmp) {
-        List<IEntityAction> entityActions = new ArrayList<IEntityAction>();
-        if (actionstmp.size() == 0)
-            return null;
-        entityActions.addAll(Arrays.asList(actionstmp.get(this.getRNG().nextInt(actionstmp.size()))));
-        return entityActions;
+    public List<IEntityAction> getAvailableActions(EntityClass entityClass) {
+        List<IEntityAction> actionstmp = new ArrayList<IEntityAction>();
+        switch (entityClass) {
+            default:
+                break;
+        }
+        return actionstmp;
+    }
+
+    public EntityClass getEntityClass() {
+        return entityclass;
+    }
+
+    protected void setupEntityClass() {
+        entityclass = EntityClass.getRandomClass();
+        this.getEntityAttribute(SharedMonsterAttributes.MAX_HEALTH).applyModifier(entityclass.getHealthModifier());
+        this.getEntityAttribute(SharedMonsterAttributes.ATTACK_DAMAGE).applyModifier(entityclass.getDamageModifier());
+        this.getEntityAttribute(SharedMonsterAttributes.MOVEMENT_SPEED).applyModifier(entityclass.getSpeedModifier());
+    }
+
+    public enum EntityClass {
```

Comment:
Maybe rename this to EntityClassType or EntityActionType or EntityFightClass.
It took me a minute to find out that it is completely unrelated to Java classes.

Could be moved in some API class as well